In [1]:
%matplotlib inline
import warnings
warnings.simplefilter('ignore')
import pandas as pd
import matplotlib.pyplot as plt

I began my analysis of the depression dataset by importing it and running some basic investigations.

In [2]:
depression_df = pd.read_csv('../data/Raw/with-depression.csv')
depression_df.shape

(6468, 4)

In [3]:
depression_df.sample()

Entity Code  Year  \
3658  Mauritius  MUS  2008   

      Prevalence - Depressive disorders - Sex: Both - Age: Age-standardized (Percent) (%)  
3658                                           3.576573

In [4]:
depression_df.columns

Index(['Entity', 'Code', 'Year',
       'Prevalence - Depressive disorders - Sex: Both - Age: Age-standardized (Percent) (%)'],
      dtype='object')

I renamed a column to make it easier to manipulate.

In [5]:
depression_df = depression_df.rename(columns={'Prevalence - Depressive disorders - Sex: Both - Age: Age-standardized (Percent) (%)':'Depression_percent'})

I continued my basic investigations.

In [6]:
depression_df.head()

Entity Code  Year  Depression_percent
0  Afghanistan  AFG  1990            4.071831
1  Afghanistan  AFG  1991            4.079531
2  Afghanistan  AFG  1992            4.088358
3  Afghanistan  AFG  1993            4.096190
4  Afghanistan  AFG  1994            4.099582

In [7]:
depression_df['Year'].unique()

array([1990, 1991, 1992, 1993, 1994, 1995, 1996, 1997, 1998, 1999, 2000,
       2001, 2002, 2003, 2004, 2005, 2006, 2007, 2008, 2009, 2010, 2011,
       2012, 2013, 2014, 2015, 2016, 2017])

In [8]:
depression_df['Entity'].nunique()

231

I imported my continents dataset.

In [9]:
continents_df = pd.read_csv('../data/Raw/countries_continents2.csv')

Then I merged it with the depression dataset. 191 of the 231 entities in the original depression dataset remain in the merged one. However, since this is close to the number of countries in the world, I am satisfied. I care less about the territories listed in the depression dataset.

In [10]:
merged_df = depression_df.merge(continents_df, left_on='Entity', right_on='Country')
merged_df['Entity'].nunique()

191

In [11]:
merged_df['Continent'].unique()

array(['Asia', 'Europe', 'Africa', 'Oceania', 'North America',
       'South America'], dtype=object)

I then pulled the mean, max, and min values of depression percent by continent.

In [12]:
merged_df.groupby('Continent')['Depression_percent'].agg(['mean', 'min','max'])

mean       min       max
Continent                                  
Africa         3.771239  2.885115  5.744704
Asia           3.376109  2.233269  5.116345
Europe         3.585069  2.139903  5.512826
North America  3.130038  2.535758  6.602754
Oceania        3.321411  2.913340  4.929038
South America  3.278590  2.194091  4.401565

In [13]:
import seaborn as sns
import plotly_express as px
import math

In [14]:
pop_df = pd.read_csv('../data/Cleaned/world_pop_long')
pop_df['Location'].unique()

array(['World', 'More developed regions', 'Less developed regions',
       'Least developed countries',
       'Less developed regions, excluding least developed countries',
       'Less developed regions, excluding China',
       'Land-locked Developing Countries (LLDC)',
       'Small Island Developing States (SIDS)', 'High-income countries',
       'Middle-income countries', 'Upper-middle-income countries',
       'Lower-middle-income countries', 'Low-income countries',
       'No income group available', 'Africa', 'Asia', 'Europe',
       'Latin America and the Caribbean', 'Northern America', 'Oceania',
       'Sub-Saharan Africa', 'Eastern Africa', 'Burundi', 'Comoros',
       'Djibouti', 'Eritrea', 'Ethiopia', 'Kenya', 'Madagascar', 'Malawi',
       'Mauritius', 'Mayotte', 'Mozambique', 'Réunion', 'Rwanda',
       'Seychelles', 'Somalia', 'South Sudan', 'Uganda', 'Tanzania',
       'Zambia', 'Zimbabwe', ' Middle Africa', 'Angola', 'Cameroon',
       'Central African Republic', 'C

After importing the long population dataset I developed in (data_analysis_Anxiety Investigation), I merged it with my depression dataset.

In [15]:
dp_df = merged_df.merge(pop_df, left_on=['Year', 'Entity'], right_on=['variable', 'Location'])
dp_df.shape

(5320, 13)

I successfully merged 189 countries into this new dataset.

In [16]:
dp_df['Entity'].nunique()

189

I then created a new variable that represented each country's actual population, since the original value was by thousand.

In [17]:
dp_df['Population']=dp_df['value']*1000

In [18]:
dp_df.head()

Entity Code  Year  Depression_percent  Unnamed: 0      Country  \
0  Afghanistan  AFG  1990            4.071831           0  Afghanistan   
1  Afghanistan  AFG  1991            4.079531           0  Afghanistan   
2  Afghanistan  AFG  1992            4.088358           0  Afghanistan   
3  Afghanistan  AFG  1993            4.096190           0  Afghanistan   
4  Afghanistan  AFG  1994            4.099582           0  Afghanistan   

          Region Continent  code     Location      Average  variable    value  \
0  Southern Asia      Asia     4  Afghanistan  24737.62069      1990  12412.0   
1  Southern Asia      Asia     4  Afghanistan  24737.62069      1991  13299.0   
2  Southern Asia      Asia     4  Afghanistan  24737.62069      1992  14486.0   
3  Southern Asia      Asia     4  Afghanistan  24737.62069      1993  15817.0   
4  Southern Asia      Asia     4  Afghanistan  24737.62069      1994  17076.0   

   Population  
0  12412000.0  
1  13299000.0  
2  14486000.0  
3  15817000.0  
4  17076000.0

Next, I created a variable to figure out the number of individuals with depression for each datapoint.

In [19]:
dp_df['Depression_population']=dp_df['Depression_percent']/100 * dp_df['Population']
dp_df.head()

Entity Code  Year  Depression_percent  Unnamed: 0      Country  \
0  Afghanistan  AFG  1990            4.071831           0  Afghanistan   
1  Afghanistan  AFG  1991            4.079531           0  Afghanistan   
2  Afghanistan  AFG  1992            4.088358           0  Afghanistan   
3  Afghanistan  AFG  1993            4.096190           0  Afghanistan   
4  Afghanistan  AFG  1994            4.099582           0  Afghanistan   

          Region Continent  code     Location      Average  variable    value  \
0  Southern Asia      Asia     4  Afghanistan  24737.62069      1990  12412.0   
1  Southern Asia      Asia     4  Afghanistan  24737.62069      1991  13299.0   
2  Southern Asia      Asia     4  Afghanistan  24737.62069      1992  14486.0   
3  Southern Asia      Asia     4  Afghanistan  24737.62069      1993  15817.0   
4  Southern Asia      Asia     4  Afghanistan  24737.62069      1994  17076.0   

   Population  Depression_population  
0  12412000.0          505395.685697  
1  13299000.0          542536.819099  
2  14486000.0          592239.575077  
3  15817000.0          647894.356027  
4  17076000.0          700044.550287

I dropped the replicated variables in the data.

In [20]:
dp3_df = dp_df.drop(['Country', 'Location','value', 'variable', 'code'], axis=1)

I renamed the average column to 'Average Pop' to make it more clear that it represented the average population of a country between 1990 and 2017.

In [21]:
dp3_df.rename(columns={'Average':'Average_pop'})

Entity Code  Year  Depression_percent  Unnamed: 0           Region  \
0     Afghanistan  AFG  1990            4.071831           0    Southern Asia   
1     Afghanistan  AFG  1991            4.079531           0    Southern Asia   
2     Afghanistan  AFG  1992            4.088358           0    Southern Asia   
3     Afghanistan  AFG  1993            4.096190           0    Southern Asia   
4     Afghanistan  AFG  1994            4.099582           0    Southern Asia   
5     Afghanistan  AFG  1995            4.104207           0    Southern Asia   
6     Afghanistan  AFG  1996            4.107500           0    Southern Asia   
7     Afghanistan  AFG  1997            4.110834           0    Southern Asia   
8     Afghanistan  AFG  1998            4.114438           0    Southern Asia   
9     Afghanistan  AFG  1999            4.117633           0    Southern Asia   
10    Afghanistan  AFG  2000            4.118610           0    Southern Asia   
11    Afghanistan  AFG  2001            4.121381           0    Southern Asia   
12    Afghanistan  AFG  2002            4.124928           0    Southern Asia   
13    Afghanistan  AFG  2003            4.125230           0    Southern Asia   
14    Afghanistan  AFG  2004            4.126384           0    Southern Asia   
15    Afghanistan  AFG  2005            4.126908           0    Southern Asia   
16    Afghanistan  AFG  2006            4.128638           0    Southern Asia   
17    Afghanistan  AFG  2007            4.129728           0    Southern Asia   
18    Afghanistan  AFG  2008            4.129656           0    Southern Asia   
19    Afghanistan  AFG  2009            4.129972           0    Southern Asia   
20    Afghanistan  AFG  2010            4.130874           0    Southern Asia   
21    Afghanistan  AFG  2011            4.130862           0    Southern Asia   
22    Afghanistan  AFG  2012            4.132485           0    Southern Asia   
23    Afghanistan  AFG  2013            4.134661           0    Southern Asia   
24    Afghanistan  AFG  2014            4.133526           0    Southern Asia   
25    Afghanistan  AFG  2015            4.135419           0    Southern Asia   
26    Afghanistan  AFG  2016            4.135694           0    Southern Asia   
27    Afghanistan  AFG  2017            4.136347           0    Southern Asia   
28        Albania  ALB  1990            2.139903           2  Southern Europe   
29        Albania  ALB  1991            2.141201           2  Southern Europe   
...           ...  ...   ...                 ...         ...              ...   
5290       Zambia  ZMB  2016            3.608843         247   Eastern Africa   
5291       Zambia  ZMB  2017            3.641811         247   Eastern Africa   
5292     Zimbabwe  ZWE  1990            3.040364         248   Eastern Africa   
5293     Zimbabwe  ZWE  1991            3.060225         248   Eastern Africa   
5294     Zimbabwe  ZWE  1992            3.077426         248   Eastern Africa   
5295     Zimbabwe  ZWE  1993            3.089601         248   Eastern Africa   
5296     Zimbabwe  ZWE  1994            3.100224         248   Eastern Africa   
5297     Zimbabwe  ZWE  1995            3.107312         248   Eastern Africa   
5298     Zimbabwe  ZWE  1996            3.105739         248   Eastern Africa   
5299     Zimbabwe  ZWE  1997            3.094765         248   Eastern Africa   
5300     Zimbabwe  ZWE  1998            3.083283         248   Eastern Africa   
5301     Zimbabwe  ZWE  1999            3.074554         248   Eastern Africa   
5302     Zimbabwe  ZWE  2000            3.069722         248   Eastern Africa   
5303     Zimbabwe  ZWE  2001            3.072727         248   Eastern Africa   
5304     Zimbabwe  ZWE  2002            3.080942         248   Eastern Africa   
5305     Zimbabwe  ZWE  2003            3.089274         248   Eastern Africa   
5306     Zimbabwe  ZWE  2004            3.096351         248   Eastern Africa   
5307     Zimbabwe  ZWE  2005            3.

I imported this cleaned depression dataset to a CSV.

In [22]:
dp3_df.to_csv('../data/Cleaned/depression_clean', index=False)